pyupgrade/_plugins/typing_pep563.py

```diff
@@ -0,0 +1,133 @@
+import ast
+import functools
+import sys
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from tokenize_rt import Offset
+
+from pyupgrade._ast_helpers import ast_to_offset
+from pyupgrade._data import register
+from pyupgrade._data import State
+from pyupgrade._data import Token
+from pyupgrade._data import TokenFunc
+
+
+def _supported_version(state: State) -> bool:
+    return (
+        state.settings.min_version >= (3, 10) or (
+            'annotations' in state.from_imports['__future__']
+        )
+    )
+
+
+def _dequote(i: int, tokens: List[Token], *, new: str) -> None:
+    tokens[i] = tokens[i]._replace(src=new)
+
+
+def _get_name(node: ast.expr) -> str:
+    if isinstance(node, ast.Name):
+        return node.id
+    elif isinstance(node, ast.Attribute):
+        return node.attr
+    else:
+        raise AssertionError(f'expected Name or Attribute: {ast.dump(node)}')
+
+
+def _process_call(node: ast.Call) -> Iterable[ast.AST]:
+    name = _get_name(node.func)
+    if name == 'TypedDict':
+        dict_ = node.args[1]
+        if isinstance(dict_, ast.Dict):
+            yield from dict_.values
+        else:
+            raise AssertionError(f'expected ast.Dict: {ast.dump(dict_)}')
```

Comment:
TypedDict has two syntaxes, one with named arguments and one with a dictionary:

`TypedDict('foo', {'k': 'int'})` and `TypedDict('foo', k='int')`